src/run/run.c

```diff
@@ -1506,9 +1506,11 @@ static int pty_forward_handler(PTYForward *f, int rcode, void *userdata) {
 
         assert(f);
 
-        if (rcode == -ECANCELED)
+        if (rcode == -ECANCELED) {
                 log_debug_errno(rcode, "PTY forwarder disconnected.");
-        else if (rcode < 0) {
+                if (!arg_wait)
+                        return sd_event_exit(c->event, EXIT_SUCCESS);
```

Comment:
well, the question is whether ^[ should be considered just "end pty forwarding" or be more treated like a ^C that hits the client, i.e. asks for program termination.

But I guess it's OK to declare that it should just end the pty forwarding, as long as ^C is then available to propelry kill the client at that point (and I think that works, right?) 

But can you add a comment to the sources here? i.e. explain this all: "If --wait is specified we'll only exit the pty fowarding, but will continue to wait for the service to end. If the user hits ^C we'll exit too."

(i mean, ideally we'd also output a friendly suggestion log message about this, but i don't see how we could make this reasonably pretty, given that most likely hitting ^[ results in an EOF/POLLHUP on the other side anyway which typically triggers termination anyway, hence any message suggesting to hit ^C is in 99.99% of all cases redundant and confusing)